example-react-app/src/i18n/LanguageSwitcher.tsx

```diff
@@ -14,9 +14,10 @@ export const LanguageSwitcher = observer((props: LanguageSwitcherProps) => {
   const mainStore = useMainStore();
 
   const handleChange = useCallback(
-    action((locale: string) => {
-      mainStore.locale = locale;
-    }),
+    () =>
+      action((locale: string) => {
+        mainStore.locale = locale;
+      }),
```

Comment:
I suspect that `mainStore.locale` won't get changed because the function returned from `action` is never invoked. Perhaps we should use `runInAction` here instead?